solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -441,6 +463,13 @@ public void doFilter(ServletRequest _request, ServletResponse _response, FilterC
       consumeInputFully(request, response);
       SolrRequestInfo.reset();
       SolrRequestParsers.cleanupMultipartFiles(request);
+
+      if (accepted) {
+        rateLimitManager.decrementActiveRequests(request);
```

Comment:
There is a race condition here where new incoming requests can get in before resume of an existing request.

Reply:
Redundant now, thanks